pyproject.toml

```diff
@@ -1,3 +1,150 @@
 [tool.black]
 line-length = 100
 include = '\.pyi?$'
+
+[tool.poetry]
+name = "pennylane"
+version = "0.35.0-dev"
```

Comment:
Are we going to maintain the version of pennylane in two spots? (Pyproject + pennylane/_version.py)?

Reply:
I'd rather not, but that was my short-term plan. can we have one read from the other?

Reply:
So apparently the Python answer is a package should not need to know it's own version within it's own code. As there shouldn't be logic within the package itself that alters behavior using the version.

Now if someone decides to use the package and has logic based on version, then they can get the package version using `importlib.metadata.version("{package}")`.

This would mean pyproject.toml being the ultimate source of truth.

Now if for backwards-compatibility reason, you want to continue to provide `package.__version__`, then yeah, you can have it read the pyproject.

I've actually done this for the Architecture team, see this: [Link](https://github.com/XanaduAI/ft-stack/blob/0c723949fde2b63a6b5e90a5c4c7a4c50b2e891a/flamingpy/_version.py)

You add `toml` as a dependency, then return the packages own version from importlib.metadata. If someone is using package in editable mode for development, then read pyproject.